2016-04-08-rules-and-queries.md

```diff
@@ -0,0 +1,342 @@
+
+### Versioning
+Now application rule supports 2 versions. Version until `2022-09-10` and from `2022-09-10`. You can specify version you want to use in [Mergado Developers](https://app.mergado.com/developres/) center by specifing version parameter. Every `date >= 2022-09-10` means, that mergado will use new version and sent product data in the new representation. 
+
+### Processing of application rule for version until 2022-09-10
+<details markdown="1">
+<summary>Version until 2022-09-10</summary>
+
+In the version `until 2022-09-10`, The backend sends the data on URL you specified in Mergado Developers center in the following format by a `HTTP POST` request:
+
+```json
+{
+    "rule_id": "12",
+    "project_id": "1",
+    "apply_log_id": "501",
+    "request_id": "1BAC92",
+    "current_format": "heureka.cz",
+    "data": [
+        {
+            "id": "123",
+            "created_at": "2016-03-27T17:18:45+00:00",
+            "updated_at": "2016-04-28T17:18:50+00:00",
+            "output_changed_at": null,
+            "data": {
+                "ITEM_ID": "890",
+                "PRODUCTNAME": "Amazon Kindle Paperwhite 3",
+                "DELIVERY_DATE": "0",
+                "EAN": "0848719006099",
+                "PRICE": "2525.10",
+                "IMGURL": "https://app.mergado.com/img/890.jpg"
+            },
+            "metadata": {
+                "..."
+            }
+        },
+        {
+            "id": "234",
+            "created_at": "2016-01-07T13:56:42+00:00",
+            "updated_at": "2016-04-28T17:18:50+00:00",
+            "output_changed_at": null,
+            "data": {
+                "ITEM_ID": "530",
+                "PRODUCTNAME": "Epson LX 100 EPS",
+                "DELIVERY_DATE": "3",
+                "EAN": "0010313601581",
+                "PRICE": "185.10",
+                "IMGURL": "https://app.mergado.com/img/530.jpg"
+            },
+            "metadata": {
+                "..."
+            }
+        }
+    ]
+}
+```
+
+Explanation of the fields:
+
+* `project_id` - ID of the project. Changes in products by rules are applied to this project.
+* `rule_id` - ID of the rule in Mergado, each rule is represented by one row in the UI.
+* `apply_log_id` - ID of the rule application's log.
+* `request_id` - ID of the request different for each request. Note that the ID doesn't change when the request is retried (see Retrying on Errors bellow).
+* `current_format` - Name of the format in which the products are represented when this rule is applied.
+* `data` - A list of products.
+    + `id` - ID of the product.
+    + `created_at` - When the product has been imported in Mergado.
+    + `updated_at` - The last time the product changed.
+    + `output_changed_at` - The last time the product changed its output values.
+    + `data` - A key-value paires containing elements, their names and values. These values are altered by rules with higher priority (applied sooner in the chain of rules).
```

Comment:
> A key-value paires containing elements

pair? buď jednotné, nebo množné

Reply:
Niesom si istý čo je tu zle. Malo by to byť množné a teda je to podľa mňa správne.